### PR TITLE
DSND-2422 Change findCompanyFilingHistory to catch DataAccessException

### PR DESCRIPTION
**Pull request by kinpang-CH**
* Change respective Service, Processor, Controller to throw and catch ServiceUnavailable
* Add missing get single transaction unit tests

### Related Jira tickets

[DSND-2422](https://companieshouse.atlassian.net/browse/DSND-2422)

### General

- [x] Is the PR as small as it can be?
- [x] Has the Jira ticket been updated with any relevant comments?
- [x] Is the `README` up to date?
- [x] Has the `POM` been updated for the latest versions of dependencies?
- [x] Has the code been double-checked against `main`?
- [x] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [x] Do the code changes have unit and integration tests with code coverage > 80%?
- [x] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

Cant be tested on any environments besides local Tilt.

[DSND-2422]: https://companieshouse.atlassian.net/browse/DSND-2422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java [hunk 1]
```diff
@@ -39,24 +39,30 @@ public Repository(final MongoTemplate mongoTemplate) {
...
+                    project()
+                            .and(ifNull("$total_count.count").then(0)).as("total_count")
+                            .andExpression("$document_list").slice(itemsPerPage, startIndex).as("document_list"));
 
-        return mongoTemplate.aggregate(aggregation, FilingHistoryDocument.class, FilingHistoryListAggregate.class)
-                .getUniqueMappedResult();
+            return mongoTemplate.aggregate(aggregation, FilingHistoryDocument.class, FilingHistoryListAggregate.class)
+                    .getUniqueMappedResult();
+        } catch (DataAccessException ex) {
+            LOGGER.error("MongoDB unavailable when locating documents for company number: %s with message: %s"
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java:61]: Log message is a bit verbose - we dont need to include company number in the message as its already in the logMap.

Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java]: Changed in latest commit to be same as the exception message.

**Review by cgregoryCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**